gui-apps/organicmaps/organicmaps-9999.ebuild

```
@@ -1,78 +1,102 @@
 
 EAPI=8
 
-PYTHON_COMPAT=( python3_{11..12} )
-inherit git-r3 python-r1 xdg cmake
+PYTHON_COMPAT=( python3_{11..14} )
+inherit git-r3 python-any-r1 xdg cmake
 EGIT_REPO_URI="https://github.com/${PN}/${PN}.git"
-# this URL is to make the tests compile since organicmaps usually dynamically clones the repo
-# maybe a better way would be to skip the test
+
 EGIT_WORLD_FEED_REPO_URI="https://github.com/${PN}/world_feed_integration_tests_data.git"
+
 # organicmaps gets more and more system libraries, we use as many
 # as currently possible, use submodules for the rest
 EGIT_SUBMODULES=(
-	3party/harfbuzz/harfbuzz
 	3party/fast_double_parser
+	3party/fast_obj
+	3party/glaze
+	3party/glfw
+	3party/harfbuzz/harfbuzz
+	3party/imgui/imgui
 	3party/just_gtfs
 	3party/protobuf/protobuf # wait for https://github.com/organicmaps/organicmaps/pull/6310
-	3party/fast_obj
 )
 
 DESCRIPTION="Offline maps and navigation using OpenStreetMap data"
 HOMEPAGE="https://organicmaps.app"
 
 LICENSE="Apache-2.0"
+
 SLOT="0"
 
-REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+BDEPEND="
+	dev-util/vulkan-headers
+	${PYTHON_DEPS}
+	test? ( dev-cpp/gtest )
+"
 
 # depend on sys-libs/zlib[minizip] when it is not pulled in as subproject anymore
 RDEPEND="
+	dev-cpp/fast_double_parser
 	dev-cpp/gflags
 	dev-db/sqlite
-	dev-lang/python
 	dev-libs/boost
 	dev-libs/icu
 	dev-libs/jansson
 	dev-libs/utfcpp
+	dev-qt/qtbase:6[gui,network,opengl,widgets]
 	dev-qt/qtpositioning:6
 	dev-qt/qtsvg:6
-	dev-util/vulkan-headers
 	media-libs/freetype
 	sys-libs/zlib
-	${PYTHON_DEPS}
 "
-DEPEND="${RDEPEND}"
 
-PATCHES=( "${FILESDIR}"/more-3party.patch "${FILESDIR}"/no-dynamic-download.patch )
+PATCHES=( "${FILESDIR}"/fix-jansson.patch "${FILESDIR}"/protobuf.patch "${FILESDIR}"/respect-ldflags.patch )
 
 WORLD_FEED_TESTS_S="${WORKDIR}/world_feed_integration_tests_data-${PV}"
 
 src_unpack () {
 	git-r3_fetch
 	git-r3_checkout
-	git-r3_fetch "${EGIT_WORLD_FEED_REPO_URI}"
-	git-r3_checkout "${EGIT_WORLD_FEED_REPO_URI}" "${WORLD_FEED_TESTS_S}"
+
+	if use test; then
+		git-r3_fetch "${EGIT_WORLD_FEED_REPO_URI}"
+		git-r3_checkout "${EGIT_WORLD_FEED_REPO_URI}" "${WORLD_FEED_TESTS_S}"
+	fi
 }
 
 src_configure() {
-	# organicmaps wants a ./configure.sh execution.
-	# However, this setups mainly stuff for Android and XCode builds that we don't need.
-	# We need just this line here
-	cp private_default.h private.h || die
-
-	CMAKE_BUILD_TYPE="RelWithDebInfo"
 	local mycmakeargs=(
 		-DWITH_SYSTEM_PROVIDED_3PARTY=yes
 		-DBUILD_SHARED_LIBS=off
-		-DTEST_DATA_REPO_URL="${WORLD_FEED_TESTS_S}"
+		-DTEST_DATA_REPO_URL="$(usex test "${WORLD_FEED_TESTS_S}" "")"
+		-DBUILD_TESTING=$(usex test ON OFF)
+		-DCMAKE_BUILD_TYPE=Release
+		-DSKIP_TESTS=1
 	)
 	cmake_src_configure
 }
 
+src_compile() {
+	CMAKE_BUILD_TARGET="desktop"
+	cmake_src_compile desktop
+}
+
+src_test() {
+	cmake_src_test
+}
+
+src_install() {
+	cmake_src_install
```

```suggestion
	dobin "${BUILD_DIR}"/OMaps
	insinto /usr/share/organicmaps
	doins -r data

	insinto /usr/share/metainfo
	doins packaging/app.organicmaps.desktop.metainfo.xml
	domenu qt/res/linux/app.organicmaps.desktop.desktop
	newicon -s 512 qt/res/logo.png organicmaps.png
```
`cmake_src_install` will build/install other components (it doesn't support installing only the desktop target).
https://github.com/organicmaps/organicmaps/blob/6baea54ad361615cf679d020ddffdc1e2d73a78f/qt/CMakeLists.txt#L135-L142

---

> `cmake_src_install` will build/install other components (it doesn't support installing only the desktop target).

But we can pass arguments?